development/tools/pkg/jobguard/filter.go

```diff
@@ -0,0 +1,48 @@
+package jobguard
+
+import "strings"
+
+// FilterStatusByName filters statuses by name
+func FilterStatusByName(in []Status, substring string) []Status {
+	var filteredStatuses []Status
+
+	for _, s := range in {
+		if !strings.Contains(s.Name, substring) {
+			continue
+		}
+
+		filteredStatuses = append(filteredStatuses, s)
+	}
+
+	return filteredStatuses
+}
+
+// FailedStatuses filters statuses by failed or error state
+func FailedStatuses(in []Status) []Status {
```

Comment:
idea to discuss: should we take into account only `required` checks?

Reply:
Hmm, does it really makes a difference? If so, in which cases?